src/app/joke.service.ts

```diff
@@ -0,0 +1,56 @@
+// joke.service.ts
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class JokeService {
+  jokeResult: string; // stores value of random joke
+  favs = []; // arr that stores list of favorite jokes
+  searchList: any = []; // arr that stores list of search results
+  catList: any = [];
+  jokeUrl = 'https://api.chucknorris.io/jokes'; // url to get random joke
+  searchUrl = 'https://api.chucknorris.io/jokes/search?query='; // url to search for jokes; requires input
+  categoryUrl = 'https://api.chucknorris.io/jokes/categories'; // url to retrieve list of categories
+  searchTerm: string; // input for search
+
+  constructor(private http: HttpClient) { }
+
+  // retrieve and display joke
+  getJoke() {
+    interface JokeTime { // work with data retrieved
+      value: string;
+      category: string;
+      id: string;
+    }
+    this.http.get<JokeTime>(this.jokeUrl + '/random').subscribe(data => {
+      console.log('Value: ' + data.value);
+      console.log('Category: ' + data.category);
+      console.log('ID: ' + data.id);
+      this.jokeResult = data.value;
+    });
+  }
+  // add currently displayed joke to the list of favorite jokes
+  addFav() {
+    this.favs.push(this.jokeResult);
+  }
+  // search for jokes
+  searchJokes(string) {
+    this.searchTerm = string;
+    this.clearJokes();
+    interface JokeSearch { // interface with search data retrieved
```

Comment:
The JokeSearch interface should not be declared inline. A better place for it would be outside the JokeService class.

Reply:
Placed interface in separate file as suggested by Nick V.